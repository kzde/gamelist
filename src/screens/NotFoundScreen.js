import React, { Fragment } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import Header from '../components/Header';
import backgroundImg from '../assets/images/rayman-legends.jpg';
import withI18N from '../contexts/withI18n';
import Link from '../components/Link';
import { colorPalettes, textSizes, media } from '../utils/styleUtils';

const Img = styled.img`
  width: 100%;
  background-size: cover;
  height: 100vh;
  background: url(${backgroundImg});
  background-repeat: no-repeat;
  background-position: center;
`;

const ContentWrapper = styled.div`
  background-color: rgba(0, 0, 0, 0.6);
  position: absolute;
  top: 40%;
  margin-left: auto;
  margin-right: auto;
  left: 0;
  right: 0;
  width: 50%;
  height: 200px;
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  font-size: ${textSizes.titleText};
  padding: 0 30px;
  ${media.desktop`font-size:${textSizes.normalText};`}
`;
const Content = styled.div`
  color: ${colorPalettes.white};
  text-align: center;
`;
const GoBackButton = styled.div`
`;

const NotFoundScreen = props => (
  <Fragment>
    <Header text={`${props.i18n._('catalog.header.title')} 404`} />
    <Img />
    <ContentWrapper>
      <Content>{props.i18n._('notfoundpage.text.pagenotfound')}</Content>
      <GoBackButton>
        <NavLink
          to="/"
          activeStyle={{
            color: colorPalettes.red
          }}
        >
          <Link text={props.i18n._('notfoundpage.text.goback')} />
        </NavLink>
      </GoBackButton>
    </ContentWrapper>
  </Fragment>
);

NotFoundScreen.propsType = {
  i18n: PropTypes.shape({
    _: PropTypes
  }).isRequired
};
export const notFoundScreen = NotFoundScreen;
export default withI18N(NotFoundScreen);
