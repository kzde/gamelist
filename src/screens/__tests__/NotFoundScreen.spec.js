import React from 'react';
import renderer from 'react-test-renderer';
import { notFoundScreen as NotFoundScreen } from '../NotFoundScreen';

jest.mock('../../components/Header', () => 'Header');
jest.mock('react-router-dom', () => ({ NavLink: 'NavLink' }));

const props = {
  i18n: {
    _: jest.fn().mockReturnValue('translation fake')
  }
};
describe('NotFoundScreen', () => {
  it('should renders correctly', () => {
    const tree = renderer.create(<NotFoundScreen {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
