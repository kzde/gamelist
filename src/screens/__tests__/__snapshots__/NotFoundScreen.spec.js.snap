// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundScreen should renders correctly 1`] = `
Array [
  <Header
    text="translation fake 404"
  />,
  <img
    className="NotFoundScreen__Img-sc-1oziblc-0 kJYjcO"
  />,
  <div
    className="NotFoundScreen__ContentWrapper-sc-1oziblc-1 dhAQha"
  >
    <div
      className="NotFoundScreen__Content-sc-1oziblc-2 itfODn"
    >
      translation fake
    </div>
    <div
      className="NotFoundScreen__GoBackButton-sc-1oziblc-3 cJwFkl"
    >
      <NavLink
        activeStyle={
          Object {
            "color": "#FE6050",
          }
        }
        to="/"
      >
        <div
          className="ClickableText__Wrapper-sc-1hvvc01-0 gReZzT"
        >
          <span
            className="ClickableText__Content-sc-1hvvc01-1 khlrPb"
            data-hover="translation fake"
          >
            translation fake
          </span>
        </div>
      </NavLink>
    </div>
  </div>,
]
`;
